autoprecompiles/src/stats_logger.rs

```diff
@@ -0,0 +1,67 @@
+use std::time::Instant;
+
+use itertools::Itertools;
+use powdr_ast::analyzed::{AlgebraicReference, PolyID, PolynomialType};
+use powdr_constraint_solver::constraint_system::ConstraintSystem;
+use powdr_number::FieldElement;
+use powdr_pilopt::qse_opt::Variable;
+
+pub struct StatsLogger {
+    start_time: Instant,
+}
+
+impl StatsLogger {
+    pub fn start<P: FieldElement>(constraint_system: &ConstraintSystem<P, Variable>) -> Self {
+        log_constraint_system_stats("Starting optimization", constraint_system);
```

Comment:
it seems this is still closely tied to the optimizer, shouldn't this be a module of it instead of autoprecompiles?
otherwise i think the type of argument (`ConstraintSystem`) and logging function should be parametrized, no?